source/MaterialXContrib/MaterialXNode/MaterialXShadingNodeImpl.cpp

```diff
@@ -79,27 +81,58 @@ MStatus bindFileTexture(MHWRender::MShaderInstance& shader,
     MHWRender::MTextureManager* textureManager = renderer ? textureManager = renderer->getTextureManager() : nullptr;
     if (textureManager)
     {
-        MaterialX::FilePath imagePath = MaterialXMaya::findInSubdirectories(searchPath, fileName);
-        if (!imagePath.isEmpty())
+        MTexture *texture = nullptr;
+        if (udimSet && !udimSet->empty())
         {
+            std::vector<int> udimTiles;
+            mx::FilePathVec udimPaths = mx::ImageHandler::getResolveUDIMInformation(fileName, *udimSet, udimTiles);
+            if (udimTiles.size() != udimPaths.size() * 2)
             {
-                VP2TextureUniquePtr texture(textureManager->acquireTexture(imagePath.asString().c_str(), MaterialX::EMPTY_STRING.c_str()));
-                if (texture)
-                {
-                    MHWRender::MTextureAssignment textureAssignment;
-                    textureAssignment.texture = texture.get();
-                    status = shader.setParameter(parameterName.c_str(), textureAssignment);
-
-                    // Get back the texture description
-                    texture->textureDescription(textureDescription);
-                }
+                throw mx::Exception("Failed to resolve UDIM information properly for file:" + fileName);
```

Comment:
super minor, may want a space after the colon

Reply:
Ah good catch. I'll fix.